### PR TITLE
[FLOC-2783] Simple API calls for Docker plugin API

### PR DESCRIPTION
**Pull request by itamarst**
1. Move a bunch of utility code out of `flocker.control.test.test_httpapi` and into `flocker.restapi.testtools`. It is not modified in any way.
2. Add basic (output) schema and API call implementations for three trivial Docker plugin API endpoints.

The relevant documentation:
https://github.com/docker/docker/blob/master/docs/extend/plugin_api.md#pluginactivate
https://github.com/docker/docker/blob/master/docs/extend/plugins_volume.md#volumedriverremove
https://github.com/docker/docker/blob/master/docs/extend/plugins_volume.md#volumedriverunmount

### CODE REVIEW

File: flocker/dockerplugin/__init__.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Copyright ClusterHQ Inc.  See LICENSE file for details.
+
+"""
+Docker plugin allowing use of Flocker to manage Docker volumes.
+
+This may eventually be a standalone package.
```
Comment (jml) [inline flocker/dockerplugin/__init__.py]: Hope springs eternal.

(No action required).

File: flocker/dockerplugin/_api.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    @_endpoint(u"Remove")
+    def volumedriver_remove(self, Name):
+        """
+        Remove a Docker volume.
+
+        :param unicode Name: The name of the volume.
+
+        In practice we don't actually delete anything. As a multi-node
+        volume driver the fact that a particular container is done with a
+        volume doesn't mean another won't elsewhere, so we don't delete
```
Comment (jml) [inline flocker/dockerplugin/_api.py]: This doesn't quite scan: "that X is done with Y doesn't mean another won't elsewhere". In particular, "won't" seems wrong. Could you please rephrase?

**Comment by jml** [comment]
Minor change requested, please address it before you merge. 

Also, please make sure the tests pass. If that requires non-trivial changes, please bounce back to me for review.

**Comment by itamarst** [comment]
Test failures look unrelated.

**Pull request merged**